dev/release/README.md

```diff
@@ -0,0 +1,116 @@
+<!---
+  Licensed to the Apache Software Foundation (ASF) under one
+  or more contributor license agreements.  See the NOTICE file
+  distributed with this work for additional information
+  regarding copyright ownership.  The ASF licenses this file
+  to you under the Apache License, Version 2.0 (the
+  "License"); you may not use this file except in compliance
+  with the License.  You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing,
+  software distributed under the License is distributed on an
+  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+  KIND, either express or implied.  See the License for the
+  specific language governing permissions and limitations
+  under the License.
+-->
+
+# Release
+
+## Overview
+
+  1. Test the revision to be released
+  2. Increment version number in `arrow/doc.go`
+  3. Prepare RC and vote (detailed later)
+  4. Publish (detailed later)
+
+### Prepare RC and vote
+
+Run `dev/release/release_rc.sh` on working copy of `git@github.com:apache/arrow-go` not your fork:
+
+```console
+$ git clone git@github.com:apache/arrow-go.git
+$ dev/release/release_rc.sh ${RC}
```

Comment:
this makes sense. As suggested below I would just go to create an apache upstream so we don't require a new clone and people can use their custom workflow without having to call the apache upstream origin, but I am fine with this

Reply:
The `apache` remote works but I think that it's a bit risky.
If we have the `apache` remote in our usual clone, we may push to the "apache" remote accidentally.

FYI: I use `https://github.com/apache/arrow.git` for `apache` remote in my kou/arrow clone. I change it to `git@github.com:apache/arrow.git` only when I need to push to the `apache` remote.

Reply:
> FYI: I use `https://github.com/apache/arrow.git` for `apache` remote in my kou/arrow clone. I change it to `git@github.com:apache/arrow.git` only when I need to push to the `apache` remote.

This is a good idea, I've updated to do the same. Thanks!